### ISSUE TITLE
Libation won't load

### ISSUE THREAD

**Issue by Ninko85** [rmcrackan/Libation#66]
Hi,
I might be doing something wrong here but nothing loads up when clicking Libation.exe
Any help please?
Thanks

**Comment by rmcrackan** [comment]
Oops. Found the bug. Try now with the [latest version](https://github.com/rmcrackan/Libation/releases/latest)

Thanks for the heads up.

**Comment by Ninko85** [comment]
Just tried version 5.3.6, still nothing loads, no errors, nothing.

**Comment by rmcrackan** [comment]
I deleted all previous versions so I could start from scratch. I downloaded 5.3.6, unzip, open that folder, and dbl-clk on Libation.exe. Windows Defender helpfully tells me this comes from the internet so I have to click "Most info" then "Run Anyway" (thank you microsoft for making my software look crazy suspicious). Then the initial "Welcome to Libation" window opens for me.

Something to try: press the windows button and type "cmd" (without quotes) and press enter. Then drag and drop Libation.exe into that window. It should show the full path to Libation. Then press enter. If it's encountering errors so early that it can't even launch, hopefully the errors will print to that cmd window.

**Comment by Ninko85** [comment]
I tried what you said, but still nothing, no errors, it just takes me back
to the command prompt.

The very first time i tried to load it up, like you I had the Windows
warning and I said run anyway. Then I get a warning from Norton security as
it's safety is unknown, again I say run every time, that's where nothing
loads. I try again and nothing happens, no warnings, no errors.

**Comment by rmcrackan** [comment]
Hmmm. Is there a place in Norton to see what programs it might be blocking?

The other thing to try is:

* Download Libation.5.3.6.zip , unzip , open folders. If you still have this on your computer, no need to re-download
* Download [New folder.zip](https://github.com/rmcrackan/Libation/files/6873009/New.folder.zip). Unzip
* Copy these 6 new debug files into Libation's folder so that they replace the original files
* Run Libation.exe

I added 5 pop up messages. Keep track of how many you see and let me know.

**Comment by Ninko85** [comment]
I'll try that shortly, sorry I've been busy recently.
Thanks for your help so far!

**Comment by gaz-beep** [comment]
Is Windows Defender is running as well as Norton? 

I've been having this too and I think its the Win Defender's uploading/real-time protection/privacy invading stuff blocking bits of Libation from running.

If it is this, there should be an error 2050 in 

Event Viewer > Applications and Services logs > Microsoft > Windows > Windows Defender > Operational

which will say which bit of Libation it objects to and which file it uploaded.

I think Win Defender is uploading the file, and either deciding to block it, or taking so long that Libation has meanwhile got stuck or whatever.  It also seems a bit random, but happens most often on the new versions (obviously) - I can still use 5.1.9 consistently, but not always newer versions.

The bad news is I don't seem to be able to stop Win Defender from actually doing this - changing the real-time settings and adding exclusions don't actually help.   So there some alarming privacy nightmares if you thought you could actually stop Defender uploading files to MS servers :)

More [here](https://superuser.com/questions/1469051/your-it-administrator-requires-a-security-scan-of-this-item-what-item).

rmcrackan - Thanks for all the work you and the others put into this, and the frequent updates.  It's very cool of you to make this.

I ran the debug code and saw messages 1 and 2 but not 3.  But I also saw different dll files in the event log at different times, and with different versions, so I'm not sure that means anything if this is the same problem!

**Comment by rmcrackan** [comment]
@gaz-beep Thank you so much for trying this. If you don't even make it to test message 3, there's a lot that won't work. That part just makes sure the Libation settings files are in order. If defender/norton won't even let Libation edit a file, there's a lot that won't work. :( If you find any fix for these, please reply here.

**Comment by gaz-beep** [comment]
@rmcrackan no problem, glad to help.  Thanks again for all the work you're putting into this.

TLDR (for someone later) I think the workaround is use an older version, and put each version in different folders (so Win Defender thinks they're different apps).  If one version doesn't start, just go one older and try again.

5.1.9 is definitely okay at the moment.

The only other thing to watch is that I ran out of disk space using one of the newer versions (on an old tablet for windows-only apps) and it was only **after** that I got the not-loading problem on that version - like Defender is doing some heuristic thing where running out of disk space was suspicious and tipped it over into deciding Libation was bad.

**Comment by rmcrackan** [comment]
That's great info to know. Thanks for posting here for the next person.

I avoid the space issue by using a dropbox folder as my books directory and keeping most of my books with 'smart sync'. This stores them online only but leaves these little pseudo files which look the same but they actually take up no hardd rive space.

Books taking up hard drive space has been a big concern with Libation. You'll be happy to know that I'm currently in the process of completely re-writing the part of the program which requires them to remain on your hard drive in order to be recognized as having been liberated. I'll be out of town for a while so I'm not pushing any big changes before I leave. After I get back it hopefully won't be too much longer and I'll have it complete. When that's done, you'll be able to liberate books then move them anywhere you want.

**Comment by gaz-beep** [comment]
No prob :)

The dropbox thing is a good idea.  I have to admit I'm not using the library management side of Libation.  I'm kind of messy so I just mark stuff as "hidden" in Libation once its downloaded and dump it all into a folder on my phone.  But your way seems a lot tidier!  That change you're working on sounds really good though.  Thank you!

**Comment by rmcrackan** [comment]
I hear ya. I used to listen to a lot of free promo books, many of which were garbage. This is why I originally created the hidden tag. Later it was also a leading factor in making filters auto-apply at start up. With both of these though the books still have to stick around or else Libation will re-download them again and again. (Hopefully) soon I'll just be able to delete the stupid things or move them to a different drive.

There's a way to delete books one at a time on amazon's site, but I keep my library as big as possible to make sure Libation can perform at volume.

**Comment by Ninko85** [comment]
Hi,
After further testing it turns out Norton was somehow blocking the program,
I fixed it by adding the Libation folder to the excluded monitoring list in
Norton.

Thanks for the help and more importantly all the hard work you continue to
do on this great program!

Thanks

**Comment by rmcrackan** [comment]
Hooray. Thanks for the update. This one had me stumped

I don’t do it for the praise but the appreciation is… well… appreciated :)
